front/lib/resources/app_resource.ts

```diff
@@ -89,6 +89,15 @@ export class AppResource extends ResourceWithVault<App> {
     });
   }
 
+  static async listByVault(auth: Authenticator, vault: VaultResource) {
+    return this.baseFetch(auth, {
+      where: {
+        vaultId: vault.id,
+        visibility: auth.isUser() ? ["public", "private"] : ["public"],
```

Comment:
Now that I merged https://github.com/dust-tt/dust/pull/7249 you want to tweak this one.